src/ui/public/timefilter/timefilter.js

```diff
@@ -20,142 +20,157 @@
+    return this.calculateBounds(this._time);
+  }
+
+  getForceNow = () => {
+    const forceNow = parseQueryString().forceNow;
+    if (!forceNow) {
+      return;
+    }
 
-uiRoutes
-  .addSetupWork(function (timefilter) {
-    return timefilter.init();
-  });
+    const ticks = Date.parse(forceNow);
+    if (isNaN(ticks)) {
+      throw new Error(`forceNow query parameter, ${forceNow}, can't be parsed by Date.parse`);
+    }
+    return new Date(ticks);
+  }
 
-uiModules
-  .get('kibana')
-  .service('timefilter', function (Private, globalState, $rootScope, config, $location) {
-    const Events = Private(EventsProvider);
+  calculateBounds = (timeRange) => {
+    return calculateBounds(timeRange, { forceNow: this.getForceNow() });
+  }
 
-    function convertISO8601(stringTime) {
-      const obj = moment(stringTime, 'YYYY-MM-DDTHH:mm:ss.SSSZ', true);
-      return obj.isValid() ? obj : stringTime;
+  getActiveBounds = () => {
+    if (this.isTimeRangeSelectorEnabled) {
+      return this.getBounds();
     }
-
-    createLegacyClass(Timefilter).inherits(Events);
-    function Timefilter() {
-      Timefilter.Super.call(this);
-
-      const self = this;
-      const diffTime = diffTimeFactory(self);
-      const diffInterval = diffIntervalFactory(self);
-
-      self.isTimeRangeSelectorEnabled = false;
-      self.isAutoRefreshSelectorEnabled = false;
-
-      self.init = _.once(function () {
-        const timeDefaults = config.get('timepicker:timeDefaults');
-        const refreshIntervalDefaults = config.get('timepicker:refreshIntervalDefaults');
-
-        // These can be date math strings or moments.
-        self.time = _.defaults(globalState.time || {}, timeDefaults);
-        self.refreshInterval = _.defaults(globalState.refreshInterval || {}, refreshIntervalDefaults);
-
-        globalState.on('fetch_with_changes', function () {
-        // clone and default to {} in one
-          const newTime = _.defaults({}, globalState.time, timeDefaults);
-          const newRefreshInterval = _.defaults({}, globalState.refreshInterval, refreshIntervalDefaults);
-
-          if (newTime) {
-            if (newTime.to) newTime.to = convertISO8601(newTime.to);
-            if (newTime.from) newTime.from = convertISO8601(newTime.from);
-          }
-
-          self.time = newTime;
-          self.refreshInterval = newRefreshInterval;
-        });
-      });
-
-      $rootScope.$$timefilter = self;
-
-      $rootScope.$watchMulti([
-        '$$timefilter.time',
-        '$$timefilter.time.from',
-        '$$timefilter.time.to',
-        '$$timefilter.time.mode'
-      ], diffTime);
-
-      $rootScope.$watchMulti([
-        '$$timefilter.refreshInterval',
-        '$$timefilter.refreshInterval.pause',
-        '$$timefilter.refreshInterval.value'
-      ], diffInterval);
+  }
+
+  /**
+   * Show the time bounds selector part of the time filter
+   */
+  enableTimeRangeSelector = () => {
+    this.isTimeRangeSelectorEnabled = true;
```

Comment:
In order for angular components to observe these mutations they need an event of some sort they can subscribe to